Automattic/wp-calypso
# Framework: bring back oauth

Step 1:
![image](https://cloud.githubusercontent.com/assets/548849/15895739/37eca926-2d8d-11e6-8ef7-3760b0fabafd.png)

Step 2:
![image](https://cloud.githubusercontent.com/assets/548849/15929324/aa639356-2e4c-11e6-86dd-41ee1253f814.png)

Step 3: Success
![image](https://cloud.githubusercontent.com/assets/548849/15929338/b80b2870-2e4c-11e6-91ec-16ff46ebb40f.png)

Step 4: Redirect (devdocs not enabled)
![image](https://cloud.githubusercontent.com/assets/548849/15929393/05ba11a8-2e4d-11e6-9438-ba285b2d9cbd.png)

Step 4: Redirect (devdocs enabled)
![image](https://cloud.githubusercontent.com/assets/548849/15962946/cc05b0e4-2f0e-11e6-9036-8fba388fbcc4.png)

This PRs adds support for oauth authentication for WP.com API to make the platform a better OSS citizen.

## To test

- Create an app in https://developer.wordpress.com/.
- To test with hostname == `calypso.localhost`: make sure the redirect URL attribute is set to `http://calypso.localhost:3000/api/oauth/token` and JS origin is set to `http://calypso.localhost:3000`.
- Edit config/development.json and add your "oauth_client_id".
- Clear localStorage and other cached entities.
- Load http://calypso.localhost:3000/

## To do

- [x] Make sure visiting the app always redirects to /authorize if there's no token.
- [x] Better welcome screen.
- [x] Better success screen.
- [x] Sometimes sites middleware is stuck when navigating back from the success redirect.
- [x] Untangle desktop code that uses oauth-login.
- [x] Set up flows with and without devdocs for developers.
